tests/fixtures/expect.go

```diff
@@ -34,52 +37,163 @@ type Expect struct {
 	gitSync       *v1alpha1.GitSync
 	gitSyncClient planepkg.GitSyncInterface
+	currentCommit string
 }
 
-// check that resources are created
-func (e *Expect) ResourcesExist(resourceType string, resources []string) *Expect {
+// check that resources are created / exist
+func (e *Expect) ResourcesExist(apiVersion, resourceType string, resources []string) *Expect {
 
 	e.t.Helper()
-	ctx := context.Background()
+	e.t.Log("Verifying that resources exist..")
+
 	for _, r := range resources {
-		result := e.kubeClient.CoreV1().RESTClient().Get().
-			Timeout(defaultTimeout).
-			Namespace(TargetNamespace).
-			Resource(resourceType).
-			Name(r).
-			Do(ctx)
-		if result.Error() != nil {
-			e.t.Fatalf("Resource %s does not exist", r)
+		if !e.doesExist(apiVersion, resourceType, r) {
+			e.t.Fatalf("Resource %s/%s does not exist", resourceType, r)
 		}
+		e.t.Logf("Resource %s/%s exists", resourceType, r)
 	}
 
 	return e
 }
 
-// check that resources are deleted
-func (e *Expect) ResourcesDontExist(resourceType string, resources []string) *Expect {
+// check that resources have been deleted or never existed
+func (e *Expect) ResourcesDontExist(apiVersion, resourceType string, resources []string) *Expect {
 
 	e.t.Helper()
+	e.t.Log("Verifying that resources no longer exist..")
+
 	for _, r := range resources {
-		if !e.isDeleted(resourceType, r) {
-			e.t.Fatalf("Resource %s not deleted", r)
+		if !e.doesNotExist(apiVersion, resourceType, r) {
+			e.t.Fatalf("Resource %s/%s not deleted", resourceType, r)
 		}
+		e.t.Logf("Resource %s/%s doesn't exist", resourceType, r)
 	}
 
 	return e
 }
 
+// verify value of resource spec to determine if change occurred or not as expected
+func (e *Expect) VerifyResourceState(apiVersion, resourceType, resource, field, key string, value interface{}) *Expect {
+
+	e.t.Helper()
+	e.t.Log("Verifying resource state is as expected..")
+
+	var (
+		err  error
+		body []byte
+	)
+
+	if !e.doesExist(apiVersion, resourceType, resource) {
+		e.t.Fatalf("Resource %s/%s does not exist", resourceType, resource)
+	}
+
+	ctx := context.Background()
+
+	if apiVersion == "v1" {
+		body, err = e.kubeClient.CoreV1().RESTClient().
+			Get().
+			Namespace(TargetNamespace).
+			Resource(resourceType).
+			Name(resource).
+			DoRaw(ctx)
+	} else {
+		body, err = e.kubeClient.CoreV1().RESTClient().Get().AbsPath(
+			fmt.Sprintf("/apis/%s/namespaces/%s/%s/%s",
+				apiVersion,
+				TargetNamespace,
+				resourceType,
+				resource)).DoRaw(ctx)
+	}
+
+	if err != nil {
+		e.t.Fatalf("Failed to verify resource %s/%s state", resourceType, resource)
+	}
+
+	object := make(map[string]interface{})
+	err = yaml.Unmarshal(body, object)
+	if err != nil {
+		e.t.Fatalf("Failed to verify resource %s/%s state", resourceType, resource)
+	}
+
+	for k, v := range object[field].(map[interface{}]interface{}) {
+		if k == key {
+			if v != value {
+				e.t.Fatalf("Resource %s/%s state is not as expected", resourceType, resource)
+			}
+		}
+	}
```

Comment:
I wonder how hard it would be to make this a little better. Right now it's basically verifying "field.key", which assumes that the value to verify is exactly 2 levels down. How hard would it be to change it so that it's just a "key" (no "field") and its value is demarcated by periods? Then you'd need to have a recursive call here.

Reply:
`w.Expect().VerifyResourceState("apps/v1", "deployments", "test-deploy", "spec.replicas", 3)`

In that case, for every "." we would just recurse further until we get to the value we want to check.

Reply:
The one complexity is that you may have someone who wants to check arrays as well like `spec.myArray[1].myKey`, so ideally it would handle arrays as well. I wonder if there's some gomodule out there which can do something like this.

Reply:
That would be helpful, something that can help iterate over a `map[string]interface{}` with many more `interface{}`

Reply:
Thanks! Looking into this

Reply:
Doesn't look like it's available as a package, I might be able to mock what they're doing to parse the YAML though.